bin/install

```diff
@@ -1,158 +1,502 @@
+    generate_bootstrap | command cat - "$fish_config_file" > "$fish_config_file.tmp"
+      or abort "Error prepending config file"
 
-  if test "$OMF_CONFIG" != "$XDG_CONFIG_HOME/omf"
-    set replacements "$replacements;s|#set|set|"
+    # Swap in the prepended file.
+    command mv "$fish_config_file.tmp" "$fish_config_file"
+      or abort "Error moving file to $fish_config_file"
+
+  # We don't have any options left, so let the user set up the bootstrap manually.
+  else
+    say "For Oh My Fish to work properly, please paste the code below into a file that Fish can run at startup:"
+    echo
+    generate_bootstrap
   end
 
-  sed  "$replacements" "$OMF_PATH/$template" > "$fish_config_file"
+  # Backup the user's theme settings. This can be removed when OMF no longer touches this file to switch themes.
+  backup_file "$FISH_CONFIG/functions/fish_prompt.fish"
 
-  report progress "Building Oh My Fish configuration..."
+  say "Building Oh My Fish configuration..."
 
+  # Set up the Oh My Fish configuration directory.
   if not test -d "$OMF_CONFIG"
-    mkdir -p "$OMF_CONFIG"
+    command mkdir -p "$OMF_CONFIG"
   end
 
-  test -f "$OMF_CONFIG/bundle"; or echo "theme default" > "$OMF_CONFIG/bundle"
-  test -f "$OMF_CONFIG/theme";  or echo "default" > "$OMF_CONFIG/theme"
+  test -f "$OMF_CONFIG/bundle";
+    or echo "theme default" > "$OMF_CONFIG/bundle"
+  test -f "$OMF_CONFIG/theme"
+    or echo "default" > "$OMF_CONFIG/theme"
 
+  # Install plugins
   fish -c "omf install"
+    or abort "Error installing plugins"
 end
 
-function main
-  report progress "Installing Oh My Fish to $OMF_PATH..."
 
-  if test -d "$OMF_PATH"
-    report error "Aborting: Existing installation detected."
+# Generates the bootstrap code used to initialize Oh My Fish on shell startup.
+function generate_bootstrap
+  echo "# Path to Oh My Fish install."
+
+  if test "$OMF_PATH" = "$OMF_PATH_DEFAULT"
+    echo "\
+set -q XDG_DATA_HOME
+  and set -gx OMF_PATH \"\$XDG_DATA_HOME/omf\"
+  or set -gx OMF_PATH \"\$HOME/.local/share/omf\""
+  else
+    echo "set -gx OMF_PATH '$OMF_PATH'"
+  end
+
+  if test "$OMF_CONFIG" != "$OMF_CONFIG_DEFAULT"
+    echo "
+# Customize Oh My Fish configuration path.
+set -gx OMF_CONFIG '$OMF_CONFIG'"
+  end
+
+  echo "
+# Load Oh My Fish configuration.
+source \$OMF_PATH/init.fish"
+end
+
+
+# Uninstalls an existing OMF installation.
+function uninstall_omf
+  is_install_dir "$OMF_PATH"
+    or abort "No installation detected at $OMF_PATH"
+
+  say "Warning: This will uninstall Oh My Fish and all plugins and themes installed in $OMF_PATH. Your configuration will not be modified. You will need to remove any Oh My Fish startup code from $FISH_CONFIG/config.fish."
+
+  confirm_yes "Are you sure you want to continue?"
+  say "Uninstalling from $OMF_PATH..."
+
+  # Remove the core framework
+  command rm -rf "$OMF_PATH/.git"
+    and command rm -r "$OMF_PATH"
```

Comment:
We should use `rm -rf` here to avoid git directory permission confirmations:

```
⋊> /p/tmp fish install                                                                                                                                14:02:08
Existing installation detected at /Users/derek/.local/share/omf
Would you like to remove the existing installation? (y/N): y
Warning: This will uninstall Oh My Fish and all plugins and themes installed in 
/Users/derek/.local/share/omf. Your configuration will not be modified. You 
will need to remove any Oh My Fish startup code from 
/Users/derek/.config/fish/config.fish.
Are you sure you want to continue? (y/N): y
Uninstalling from /Users/derek/.local/share/omf...
override r--r--r--  derek/staff for /Users/derek/.local/share/omf/pkg/apt/.git/objects/07/70011cff189b773de7eeedce44cb60ab6c2931?
```